File: src/C3.Infrastructure/Preferences/LegacySettingsProfileCandidate.vb
Imports System.IO

Namespace Preferences

    ''' <summary>
    ''' Identifies one legacy ApplicationSettingsBase profile without opening it.
    ''' </summary>
    Public NotInheritable Class LegacySettingsProfileCandidate

        Public Sub New(filePath As String, profileVersion As Version, lastWriteTimeUtc As DateTime)
            If String.IsNullOrWhiteSpace(filePath) Then
                Throw New ArgumentException("A settings profile path is required.", "filePath")
            End If
            If profileVersion Is Nothing Then
                Throw New ArgumentNullException("profileVersion")
            End If

            Me.FilePath = Path.GetFullPath(filePath)
            Me.ProfileVersion = profileVersion
            Me.LastWriteTimeUtc = lastWriteTimeUtc.ToUniversalTime()
        End Sub

        Public ReadOnly Property FilePath As String

        Public ReadOnly Property ProfileVersion As Version

        Public ReadOnly Property LastWriteTimeUtc As DateTime

    End Class

End Namespace

File: src/C3.Infrastructure/Preferences/LegacySettingsProfileCandidate.cs
using System;
using System.IO;
using System.Runtime.CompilerServices;

namespace C3.Infrastructure.Preferences
{
	public sealed class LegacySettingsProfileCandidate
	{
		public string FilePath
		{
			get;
		}

		public Version ProfileVersion
		{
			get;
		}

		public DateTime LastWriteTimeUtc
		{
			get;
		}

		public LegacySettingsProfileCandidate(string filePath, Version profileVersion, DateTime lastWriteTimeUtc)
		{
			if (string.IsNullOrWhiteSpace(filePath))
			{
				throw new ArgumentException("A settings profile path is required.", "filePath");
			}
			if ((object)profileVersion == null)
			{
				throw new ArgumentNullException("profileVersion");
			}
			FilePath = Path.GetFullPath(filePath);
			ProfileVersion = profileVersion;
			LastWriteTimeUtc = lastWriteTimeUtc.ToUniversalTime();
		}
	}
}
